pkg/crypto/wallet/wallet_api.go

```diff
@@ -56,6 +66,19 @@ type Crypto interface {
 	//
 	//		in case of error, the index will be -1
 	FindVerKey(candidateKeys []string) (int, error)
+
+	// ConvertToEncryptionKey will create and persist a Curve25519 encryption keypair
+	// created from the ed25519 keypair with the provided public key.
+	//
+	// Returns:
+	//
+	// []byte: The public key of the encryption keypair created. nil in case of error.
+	//
+	// error:
+	ConvertToEncryptionKey(pubkey []byte) ([]byte, error)
```

Comment:
this function is specific to legacy crypto, it should not be part of this interface

Reply:
I thought this would be removed by your PR, since keypairs in the wallet won't need to be converted on the fly (by this method) any more, since the Signing and Encryption KeyPairs are going to be generated and persisted together.

Reply:
I've made it private, and it's now called inside CreateSigningKey, so encryption keys are available when needed.